packages/xdl/src/Project.js

```diff
@@ -303,6 +304,96 @@ export async function getLatestReleaseAsync(
   }
 }
 
+// Takes multiple exported apps in sourceDirs and coalesces them to one app in outputDir
+export async function mergeAppDistributions(
+  projectRoot: string,
+  sourceDirs: Array<string>,
+  outputDir: string
+) {
+  const assetPathToWrite = path.resolve(projectRoot, path.join(outputDir, 'assets'));
+  await fs.ensureDir(assetPathToWrite);
+  const bundlesPathToWrite = path.resolve(projectRoot, path.join(outputDir, 'bundles'));
+  await fs.ensureDir(bundlesPathToWrite);
+
+  // merge files from bundles and assets
+  const androidIndexes = [];
+  const iosIndexes = [];
+
+  for (let sourceDir of sourceDirs) {
+    const promises = [];
+
+    // copy over assets/bundles from other src dirs to the output dir
+    if (sourceDir !== outputDir) {
+      // copy file over to assetPath
+      const sourceAssetDir = path.resolve(projectRoot, path.join(sourceDir, 'assets'));
+      const outputAssetDir = path.resolve(projectRoot, path.join(outputDir, 'assets'));
+      const assetPromise = fs.copy(sourceAssetDir, outputAssetDir);
+      promises.push(assetPromise);
+
+      // copy files over to bundlePath
+      const sourceBundleDir = path.resolve(projectRoot, path.join(sourceDir, 'bundles'));
+      const outputBundleDir = path.resolve(projectRoot, path.join(outputDir, 'bundles'));
+      const bundlePromise = fs.copy(sourceBundleDir, outputBundleDir);
+      promises.push(bundlePromise);
+
+      await Promise.all(promises);
+    }
+
+    // put index.jsons into memory
+    const putJsonInMemory = async (indexPath, accumulator) => {
+      const index = await new JsonFile(indexPath).readAsync();
```

Comment:
You can also use `JsonFile` without creating an instance:
```suggestion
      const index = await JsonFile.readAsync(indexPath);
```